xhtml2pdf/w3c/cssSpecial.py

```diff
@@ -242,7 +242,7 @@ def parseSpecialRules(declarations, debug=0):
             part = getNextPart(parts) or oparts
             if part:
 
-                if hasattr(part, '__iter__') and (type("." in part) or ("data:" in part)):
+                if hasattr(part, '__iter__') and (("." in part) or ("data:" in part)):
                     dd.append(("background-image", part, last))
```

Comment:
Just to let you know:

`hasattr(part, '__iter__')` will fail in Python < 3.0 if `part` is a string. This is because strings in Python 2.7 don't have an `iter` attribute. As far as I can tell,  `part`can only be a string (or an object called "NotImplemented" if there is no attribute for `background:`) and not a list or tuple. That's why I've changed it to `isinstance(part, six.string_type)` .

You don't have to change it, as it's already part of my PR. It's just some information I wanted to share. 😄

Reply:
Yes I know, but I changed it to enable the merge, as soon as the merge is completed you will able to complete your PR with out testrender issues